docs/examples/mht_example.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+========================================================
+General Multi Hypotheses tracking implementation example
+========================================================
+"""
+
+# %%
+# The multi hypotheses tracking (MHT) algorithm is considered one of the best tracking algorithms,
+# consisting of creating a tree of potential tracks for
+# each target candidate (in a multi-target scenario) and pruning such hypotheses
+# in the data association phase. It is particularly efficient in maintaining trajectories of
+# multiple objects and handling uncertainties and ambiguities of tracks (e.g. presence of
+# clutter).
+#
+# MHT, by definition, has several algorithms that fall under this definition, which
+# include Global Nearest Neighbour
+# (GNN, tutorial `here <https://stonesoup.readthedocs.io/en/v1.1/auto_tutorials/06_DataAssociation-MultiTargetTutorial.html#sphx-glr-auto-tutorials-06-dataassociation-multitargettutorial-py>`__),
```

Comment:
Should use internal doc references e.g. `` :ref:`auto_tutorials/06_DataAssociation-MultiTargetTutorial` ``